### PR TITLE
homework4

### PR DESCRIPTION
**Pull request by zadziarnouksi-taras**
For checking!

### CODE REVIEW

File: homework4/degree.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * */
+
+void check(int *x);
+int exponentiation(int x, int n);
+
+int main() {
+    int x, n;
+
+    cout << "Please, enter the number you want to raise to the power: ";
+    cin >> x;
```
Comment (VadzimAlkhimionak) [inline homework4/degree.cpp:16]: create function for input

File: homework4/degree.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    check(&x);
+
+    cout << "Enter the degree to which you want to raise the number: ";
+    cin >> n;
```
Comment (VadzimAlkhimionak) [inline homework4/degree.cpp:21]: create function for input

File: homework4/degree.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        return 1;
+    } else if (x != 0 and n == 1) {
+        return x;
+    } else {
+        if (n == 0) {
+            return 1;
+        } else {
+            return x * exponentiation(x, n - 1);
+        }
+    }
```
Comment (VadzimAlkhimionak) [inline homework4/degree.cpp:44]: not only one action in this function

File: homework4/units.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+
+using namespace std;
+
+/**
+ * units.cpp: UNITS. Find all natural numbers up to m whose binary notation consists of ones.
+ * */
+
+void check(int *x);
+bool isConditionSatisfied(int x);
```
Comment (VadzimAlkhimionak) [inline homework4/units.cpp:10]: good name of function!

File: homework4/units.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+int main() {
+    int x;
+
+    cout << "Please, enter the number: ";
+    cin >> x;
```
Comment (VadzimAlkhimionak) [inline homework4/units.cpp:17]: create function for input

File: homework4/units.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    check(&x);
+
+    cout << "Result is ";
+
+    for (int i = 1; i < x; i++) {
+        //isConditionSatisfied(i): true - if the number 'i' whose binary notation consists of only ones, else - false.
```
Comment (VadzimAlkhimionak) [inline homework4/units.cpp:24]: delete useless comments

**Review by VadzimAlkhimionak — COMMENTED** [review]
I left some comments. Fix them. Good work! Well done

File: homework4/degree.cpp (hunk 1 above)
Comment (zadziarnouksi-taras) [reply homework4/degree.cpp]: Now I will definitely remember.

File: homework4/degree.cpp (hunk 2 above)
Comment (zadziarnouksi-taras) [reply homework4/degree.cpp]: Checked.

File: homework4/degree.cpp (hunk 3 above)
Comment (zadziarnouksi-taras) [reply homework4/degree.cpp]: `if (x == 0) {
        result = 0;
    } else if (x != 0 and n == 0) {
        result = 1;
    } else if (x != 0 and n == 1) {
        result = x;
    } else {
        result = exponentiation(x, n);
    }`
and function exponentiation(x, n):
`int exponentiation(int x, int n) {
    if (n == 0) {
        return 1;
    } else {
        return x * exponentiation(x, n - 1);
    }
}`

Comment (zadziarnouksi-taras) [reply homework4/degree.cpp]: I agree, confused

File: homework4/units.cpp (hunk 5 above)
Comment (zadziarnouksi-taras) [reply homework4/units.cpp]: Thanks :)

File: homework4/units.cpp (hunk 6 above)
Comment (zadziarnouksi-taras) [reply homework4/units.cpp]: Got it!

File: homework4/units.cpp (hunk 4 above)
Comment (zadziarnouksi-taras) [reply homework4/units.cpp:13]: Thanks :)

File: homework4/units.cpp (hunk 5 above)
Comment (zadziarnouksi-taras) [reply homework4/units.cpp]: I will never forget again ;)

**Pull request merged**